client/my-sites/pages/page/index.js

```diff
@@ -203,6 +204,33 @@ class Page extends Component {
 		);
 	}
 
+	setFrontPage() {
+		{
+			alert( 'This feature is still being developed!' );
+		}
+	}
+
+	getFrontPageItem() {
+		if ( this.props.hasStaticFrontPage && this.props.isPostsPage ) {
+			return null;
+		}
+
+		if ( ! utils.userCan( 'edit_post', this.props.page ) ) {
+			return null;
+		}
+
+		if ( ! isEnabled( 'manage/pages/set-front-page' ) ) {
+			return null;
+		}
+
+		return (
+			<PopoverMenuItem onClick={ this.setFrontPage }>
+				<Gridicon icon="house" size={ 18 } />
+				{ this.props.translate( 'Set as Front Page' ) }
+			</PopoverMenuItem>
+		);
```

Comment:
Following @folletto's [original design](https://github.com/Automattic/wp-calypso/pull/24087#issuecomment-381099000), we should probably add a separator before this menu item, like we do for the trash item below.

Something like this (untested code):
```jsx
		return [
			<MenuSeparator key="separator" />,
			<PopoverMenuItem key="item" onClick={ this.setFrontPage }>
				<Gridicon icon="house" size={ 18 } />
				{ this.props.translate( 'Set as Front Page' ) }
			</PopoverMenuItem>
		];
```